src/context/ApplicationContext.jsx

```diff
@@ -0,0 +1,48 @@
+import { useState, useContext, createContext, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { UrlSearchParamBuilder } from '../utils/api-utils/UrlSearchParamBuilder';
+import { getTimePeriodTypes } from '../api/services/timecardService';
+
+export const ApplicationContext = createContext();
+
+const fetchTimePeriodTypesData = async () => {
+  try {
+    const params = new UrlSearchParamBuilder()
+      .setTenantId('00000000-0000-0000-0000-000000000000')
+      .getUrlSearchParams();
+    const response = await getTimePeriodTypes(params);
+    if (response.data?.items) {
+      return response.data.items;
+    }
+  } catch (error) {
+    console.error(error);
+    return [];
```

Comment:
We’re yet to establish our wider strategy for handling unexpected errors but I’m wondering about this bit.

If we don’t get timeperiodtypes are we able to use the timecard functionality? So is there any value in returning an empty array?

If not, I’d be tempted to stop catching this error and let it bubble up. It’ll make it more obvious that we’ve got work to do to handle random errors 😄

Reply:
We added this in as I didn't like the idea of letting the error just happen, but I could add a comment here similar to EditShiftHours with a TODO saying that error handling is to be done when the work is done on handling unexpected errors?

Reply:
As discussed, I've removed this catch statement now